### ISSUE TITLE
[BUG] Cannot customize memory request/limit on `instance-manager` and `replica-manager`

### ISSUE THREAD

**Issue by fzyzcjy** [longhorn/longhorn#4009]
Hi thanks for the lib! However I want to customize memory request/limit on `instance-manager` and `replica-manager`.

https://longhorn.io/docs/1.2.4/references/settings/ does have CPU requests, but seems no memory request/limits?

**Comment by derekbit** [comment]
@fzyzcjy
Can you run `kubectl -n longhorn-system get settings.longhorn.io` to see if the CPU limit are set correctly?

**Comment by fzyzcjy** [comment]
It is unrelated to the question...

**Comment by derekbit** [comment]
Ah, sorry, my bad. Misunderstood your question.
Longhorn does not support memory request/limits setting right now.

**Comment by PhanLe1010** [comment]
Maybe this is a feature request instead of a bug?

**Comment by fzyzcjy** [comment]
If so, then it seems to be a FR.

**Comment by fzyzcjy** [comment]
I guess it should not be very hard to implement - just add a sibling to CPU limit :)

**Comment by PhanLe1010** [comment]
We also have to be careful about the upgrade cases, modifying those settings may lead to crashing instance manager pods.

I am also wondering if there is a reason that Longhorn doesn't have this setting before. cc @shuo-wu @joshimoo

**Comment by atze234** [comment]
Any updates here?

**Comment by barrelmaker97** [comment]
Any updates on this one? Now that we've had releases with memory leaks, this seems quite relevant:
https://github.com/longhorn/longhorn/issues/12573

**Comment by derekbit** [comment]
> Any updates on this one? Now that we've had releases with memory leaks, this seems quite relevant: [#12573](https://github.com/longhorn/longhorn/issues/12573)

@barrelmaker97 
Have you checked https://longhorn.io/docs/1.11.0/important-notes/#longhorn-instance-manager-image?

**Comment by barrelmaker97** [comment]
Yes, applied the hotfix today! But it would be nice if we could use the tools that Kubernetes provides in order to prevent a memory leak from turning into a much bigger issue.

**Comment by derekbit** [comment]
> Yes, applied the hotfix today! But it would be nice if we could use the tools that Kubernetes provides in order to prevent a memory leak from turning into a much bigger issue.

Yes, we will introduce the resource constraint. Please follow up https://github.com/longhorn/longhorn/issues/6351. Thank you.

**Comment by derekbit** [comment]
Duplicated https://github.com/longhorn/longhorn/issues/6351
